File: AtCoder/ACP/BootCamp4b/Hard100_2/cpp/C_Digits_in_Multiplication.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef pair<ll, ll> pll;

ll N;

vector<pll> divisors(ll n) {
  vector<pll> res;
  for (ll i = 1; i * i <= n; ++i) {
    if (n % i == 0) {
        res.push_back({i, n / i});
    }
  }
  return res;
}

void solve() {
  vector<pll> V = divisors(N);
  ll res = 10;
  for (auto p : V) {
    res = min(res, (ll)max(to_string(p.first).size(), to_string(p.second).size()));
  }
  printf("%lld\n", res);
}

int main() {
  cin >> N;
  solve();
}

File: AtCoder/ACP/BootCamp4b/Hard100_2/python/C_Digits_in_Multiplication.py
N = int(input())


def divisors(n):
    res = []
    for i in range(1, int(n**0.5) + 1):
        if n % i == 0:
            res.append((i, n // i))
    return res


V = divisors(N)
res = 10
for x, y in V:
    res = min(res, max(len(str(x)), len(str(y))))
print(res)
